sa/model.go

```diff
@@ -396,3 +400,152 @@ func modelToOrder(om *orderModel) (*corepb.Order, error) {
 	}
 	return order, nil
 }
+
+var challTypeToBit = map[string]uint{
+	"http-01":     0,
+	"tls-sni-01":  1,
+	"dns-01":      2,
+	"tls-alpn-01": 3,
+}
+
+var bitToChallType = map[uint]string{
+	0: "http-01",
+	1: "tls-sni-01",
+	2: "dns-01",
+	3: "tls-alpn-01",
+}
+
+type authz2Model struct {
+	ID               int64
+	IdentifierType   string
+	IdentifierValue  string
+	RegistrationID   int64
+	Status           string
+	Expires          *time.Time
+	Challenges       byte
+	Attempted        string
+	Token            []byte
+	ValidationError  []byte
+	ValidationRecord []byte
+}
+
+func authzPBToModel(authz *corepb.Authorization) (*authz2Model, error) {
+	if authz.V2 == nil || !*authz.V2 {
+		return nil, errors.New("authorization is not v2 format")
+	}
+	expires := time.Unix(0, *authz.Expires)
+	id, err := strconv.Atoi(*authz.Id)
+	if err != nil {
+		return nil, err
+	}
+	am := &authz2Model{
+		ID:              int64(id),
+		IdentifierValue: *authz.Identifier,
+		RegistrationID:  *authz.RegistrationID,
+		Status:          *authz.Status,
+		Expires:         &expires,
+	}
+	var tokenStr string
+	for _, chall := range authz.Challenges {
```

Comment:
I think this block of code could use some comments to describe the high-level of what's going on here & why.